From 962cb46a78d9208ec6dfb307a9f41c8b0ce09d87 Mon Sep 17 00:00:00 2001
From: Harry Wentland <harry.wentland@amd.com>
Date: Thu, 4 Jan 2018 11:01:22 -0500
Subject: [PATCH 4578/5855] drm/amd/powerplay: Fix no-cast warnings in
 smu7_smu_fini
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

This fixes these warnings:

drivers/gpu/drm/amd/amdgpu/../powerplay/smumgr/smu7_smumgr.c: In function ‘smu7_smu_fini’:
drivers/gpu/drm/amd/amdgpu/../powerplay/smumgr/smu7_smumgr.c:653:33: warning: passing argument 2 of ‘smu_free_memory’ makes pointer from integer without a cast [-Wint-conversion]
  smu_free_memory(hwmgr->device, smu_data->header_buffer.handle);
                                 ^~~~~~~~
In file included from drivers/gpu/drm/amd/amdgpu/../powerplay/smumgr/smu7_smumgr.c:26:0:
drivers/gpu/drm/amd/amdgpu/../powerplay/inc/smumgr.h:114:12: note: expected ‘void *’ but argument is of type ‘long unsigned int’
 extern int smu_free_memory(void *device, void *handle);
            ^~~~~~~~~~~~~~~
drivers/gpu/drm/amd/amdgpu/../powerplay/smumgr/smu7_smumgr.c:655:34: warning: passing argument 2 of ‘smu_free_memory’ makes pointer from integer without a cast [-Wint-conversion]
   smu_free_memory(hwmgr->device, smu_data->smu_buffer.handle);
                                  ^~~~~~~~
In file included from drivers/gpu/drm/amd/amdgpu/../powerplay/smumgr/smu7_smumgr.c:26:0:
drivers/gpu/drm/amd/amdgpu/../powerplay/inc/smumgr.h:114:12: note: expected ‘void *’ but argument is of type ‘long unsigned int’
 extern int smu_free_memory(void *device, void *handle);
            ^~~~~~~~~~~~~~~

Reviewed-by: Rex Zhu <Rex.Zhu@amd.com>
Acked-by: Alex Deucher <alexander.deucher@amd.com>
Signed-off-by: Harry Wentland <harry.wentland@amd.com>
---
 drivers/gpu/drm/amd/powerplay/smumgr/smu7_smumgr.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/gpu/drm/amd/powerplay/smumgr/smu7_smumgr.c b/drivers/gpu/drm/amd/powerplay/smumgr/smu7_smumgr.c
index 25dd778..311ff37 100644
--- a/drivers/gpu/drm/amd/powerplay/smumgr/smu7_smumgr.c
+++ b/drivers/gpu/drm/amd/powerplay/smumgr/smu7_smumgr.c
@@ -650,9 +650,9 @@ int smu7_smu_fini(struct pp_hwmgr *hwmgr)
 {
 	struct smu7_smumgr *smu_data = (struct smu7_smumgr *)(hwmgr->smu_backend);
 
-	smu_free_memory(hwmgr->device, smu_data->header_buffer.handle);
+	smu_free_memory(hwmgr->device, (void *) smu_data->header_buffer.handle);
 	if (!cgs_is_virtualization_enabled(hwmgr->device))
-		smu_free_memory(hwmgr->device, smu_data->smu_buffer.handle);
+		smu_free_memory(hwmgr->device, (void *) smu_data->smu_buffer.handle);
 
 	kfree(hwmgr->smu_backend);
 	hwmgr->smu_backend = NULL;
-- 
2.7.4

